packages/blocks/src/root-block/edgeless/components/color-picker/utils.ts

```diff
@@ -109,69 +88,166 @@ export const rgbToHsv = ({ r, g, b }: Rgb): Hsv => {
 
   h /= 6;
 
-  h *= 360;
-  s *= 100;
-  v *= 100;
-
   return { h, s, v };
 };
 
 // Converts an HSV color value to RGB.
 export const hsvToRgb = ({ h, s, v }: Hsv): Rgb => {
-  h = bound01(h, 360) * 6;
-  s = bound01(s, 100);
-  v = bound01(v, 100);
+  if (h < 0) h = (h + 1) % 1; // wrap
+  h *= 6;
+  s = clamp(0, s, 1);
 
   const i = Math.floor(h),
     f = h - i,
     p = v * (1 - s),
     q = v * (1 - f * s),
     t = v * (1 - (1 - f) * s),
-    mod = i % 6;
+    m = i % 6;
+
+  let rgb = [0, 0, 0];
 
-  let r = [v, q, p, p, t, v][mod],
-    g = [t, v, v, q, p, p][mod],
-    b = [p, p, t, v, v, q][mod];
+  if (m === 0) rgb = [v, t, p];
+  if (m === 1) rgb = [q, v, p];
+  if (m === 2) rgb = [p, v, t];
+  if (m === 3) rgb = [p, q, v];
+  if (m === 4) rgb = [t, p, v];
+  if (m === 5) rgb = [v, p, q];
 
-  r *= 255;
-  g *= 255;
-  b *= 255;
+  const [r, g, b] = rgb;
 
   return { r, g, b };
 };
 
+// Converts a RGBA color value to HSVA .
+export const rgbaToHsva = (rgba: Rgba): Hsva => ({
+  ...rgbToHsv(rgba),
+  a: rgba.a,
+});
+
+// Converts an HSVA color value to RGBA.
+export const hsvaToRgba = (hsva: Hsva): Rgba => ({
+  ...hsvToRgb(hsva),
+  a: hsva.a,
+});
+
 // Converts an RGB color to hex
 export const rgbToHex = ({ r, g, b }: Rgb) =>
-  [r, g, b].map(n => Math.round(n).toString(16).padStart(2, '0')).join('');
+  [r, g, b]
+    .map(n => n * 255)
+    .map(Math.round)
+    .map(s => s.toString(16).padStart(2, '0'))
+    .join('');
 
-// Converts an HSV color to CSS's `hex8`.
-export const rgbaToHex8 = ({ r, g, b }: Rgb, a = 1) =>
-  [r, g, b, a * 255]
-    .map(n => Math.round(n).toString(16).padStart(2, '0'))
+// Converts an RGBA color to CSS's hex8 string.
+export const rgbaToHex8 = ({ r, g, b, a }: Rgba) => {
+  const hex = [r, g, b, a ?? 1]
+    .map(n => n * 255)
+    .map(Math.round)
+    .map(n => n.toString(16).padStart(2, '0'))
     .join('');
+  return `#${hex}`;
+};
 
 // Converts an HSV color to CSS's `rgba`.
-export const hsvToRgbaString = (color: Hsv, a = 1) => {
-  const { r, g, b } = hsvToRgb(color);
-  return `rgba(${r},${g},${b},${a})`;
+export const hsvToHex8 = (color: Hsv, a = 1) =>
+  rgbaToHex8({ ...hsvToRgb(color), a });
+
+// Parses an hex string to RGBA.
+export const parseHexToRgba = (hex: string) => {
```

Comment:
What if the `hex` passed in is not a valid hex value? Seems you didn't consider how to handle errors when transforming colors. This will lead to runtime errors.

Reply:
In `#editHex` method, user input has been checked.